clang/lib/Sema/SemaChecking.cpp

```diff
@@ -16538,6 +16538,21 @@ void Sema::DiagnoseAlwaysNonNullPointer(Expr *E,
     }
   }
 
+  // Complain if we are converting a lambda expression to a boolean value
+  if (const auto *MCallExpr = dyn_cast<CXXMemberCallExpr>(E)) {
+    if (const auto *MRecordDecl = MCallExpr->getRecordDecl();
+        MRecordDecl && MRecordDecl->isLambda()) {
+      std::string Str;
+      llvm::raw_string_ostream S(Str);
+
+      E->printPretty(S, nullptr, getPrintingPolicy());
+      Diag(E->getExprLoc(), diag::warn_impcast_pointer_to_bool)
+          << /*LambdaPointerConversionOperatorType*/ 3 << S.str()
+          << MRecordDecl->getSourceRange() << Range << IsEqual;
```

Comment:
```suggestion
      Diag(E->getExprLoc(), diag::warn_impcast_pointer_to_bool)
          << /*LambdaPointerConversionOperatorType*/ 3
          << MRecordDecl->getSourceRange() << Range << IsEqual;
```
This way we don't pretty print the entire lambda body in the diagnostic message.

Reply:
Whoops, I accidentally missed that! Thanks for pointing out the mistakes. Since we don't use `S` anyway, can I omit the call to `printPretty()` altogether?